src/components/Button.svelte

```diff
@@ -7,81 +12,129 @@
     | "secondary-white"
     | "secondary-custom";
 
+  export let size: "small" | "medium" | "large" = "medium";
+
   // backgroundColor and textColor props only applied on custom button types
   export let backgroundColor = "#fff";
   export let textColor = "var(--text-dark)";
 
   // append hover-animated arrow
-  export let arrow: boolean | undefined = false;
-</script>
+  export let arrow = false;
 
-<button
-  class="button-{type}{arrow ? ' arrow' : ''}"
-  on:click
-  style={type.endsWith("custom")
+  $: customStyles = type.endsWith("custom")
     ? `background-color: ${
         type.startsWith("primary") ? backgroundColor : "transparent"
       }; color: ${textColor}; border-color: ${
         type.startsWith("primary") ? backgroundColor : textColor
       }`
-    : undefined}
+    : undefined;
+  $: classes = `button ${type} ${size} ${arrow ? "arrow" : ""}`;
+
+  $: tag = href ? "a" : "button";
+  $: props = {
+    href,
+    class: classes,
+    style: customStyles,
+  };
+  $: isExternal =
+    href?.startsWith("http") ||
+    href?.startsWith("mailto") ||
+    href?.startsWith("tel");
+</script>
+
+<svelte:element
+  this={tag}
+  {...props}
+  on:click
+  sveltekit:prefetch={!isExternal ? true : undefined}
```

Comment:
Really wanna update SvelteKit for this: https://kit.svelte.dev/docs/link-options#data-sveltekit-prefetch

I might go for it soon, SvelteKit is now in release candidate phase so they won't push anything causing an entire repo refactor again.